mock_tests/test_auth.py

```diff
@@ -219,3 +218,18 @@ def test_with_simple_auth_no_oidc(weaviate_mock, recwarn):
 
     weaviate_mock.check_assertions()
     assert len(recwarn) == 0
+
+
+def test_with_simple_auth_no_oidc_via_additional_headers(weaviate_mock, recwarn):
+    weaviate_mock.expect_request(
+        "/v1/schema",
+        headers={"Authorization": "Bearer " + "Super-secret-key"},
```

Comment:
While implementing this test I noticed that the `headers` parameter in the `expect_request` method has no effect with a Bearer token as Authorization (also in the other tests). The reason for that is that the following:

- The [expect_request](https://github.com/csernazs/pytest-httpserver/blob/master/pytest_httpserver/httpserver.py#L945) method of `pytest_httpserver` uses some header matching logic (see [here](https://github.com/csernazs/pytest-httpserver/blob/master/pytest_httpserver/httpserver.py#L152-L154)):
   ```python
     HeaderValueMatcher.DEFAULT_MATCHERS = defaultdict(
       lambda: HeaderValueMatcher.default_header_value_matcher,
       {"Authorization": HeaderValueMatcher.authorization_header_value_matcher},
    )
    ```
-  The mentioned `authorization_header_value_matcher` method compares the expected and actual value by calling the  [parse_authorization_header](https://github.com/csernazs/pytest-httpserver/blob/master/pytest_httpserver/httpserver.py#L136) of werkzeug:
      ```python
    return parse_authorization_header(actual) == parse_authorization_header(expected)
    ```
- This function is as of `werkzeug 2.2.3` in turn isn't capable of detecting Bearer tokens and hence just returns `None` for the actual and expected keyword

Two possible solutions:
- Passing an other matcher to the `expect_request` method so that the `authorization_header_value_matcher` is not used. For example: 
  ```python
      from pytest_httpserver import HeaderValueMatcher
      httpserver.expect_request("/", headers={"Authorization": "Bearer " + "Super-secret-key"}, header_value_matcher=HeaderValueMatcher.default_header_value_matcher).respond_with_json({})
  ```
- Wait until `werkzeug 2.3` is released and `pytest_httpserver` uses it as the `parse_authorization_header` is deprecated and the new implementation is capable of detecting Bearer tokens. But don't know when this will happen.

Reply:
oh, that's a good find.

I just tried option 1 and it doesn't seem to work for me, I assume because because requests adds a bunch of extra headers.

I think we could add our [own header matcher](https://pytest-httpserver.readthedocs.io/en/latest/howto.html#advanced-header-matching) that makes sure that all expected headers are part of the actual header and ignores the rest. Would you like to give it a try? Otherwise I could also do it

Reply:
ah, true. my example doesn't work. 

An own header matcher would work, I can also give it a try. 

Another way I just saw in the link you sent is to overwrite the default matchers:
```python
from pytest_httpserver import HeaderValueMatcher
HeaderValueMatcher.DEFAULT_MATCHERS["Authorization"] = HeaderValueMatcher.default_header_value_matcher
```

What do you think about that? That's just a one-liner

Reply:
one-liner would be perfect :)

Maybe we could put that into the conftest.py file, then it should apply to all mock-tests?

Reply:
👍 added also a comment why it is done

Reply:
great, thank you!